test/backend/routes.spec.js

```diff
@@ -319,10 +323,37 @@ describe('Authentication - ', () => {
                                  'https://bad-domain.plot.ly/v2/users/current failed, ' +
                                  'reason: getaddrinfo ENOTFOUND ' +
                                  'bad-domain.plot.ly bad-domain.plot.ly:443'));
-                    done();
-                })).catch(done);
+                    return;
+                }));
             }, 2000);
-        });
+            done();
+        }).catch(done);
+    });
+
+    it('renews access-token if expired', function(done) {
+        saveSetting('ALLOWED_USERS', [username]);
+
+
+        // set access-token expiry of 3 sec:
+        accessTokenExpiryStub.returns(3);
+
+        POST('oauth2', {access_token: accessToken})
+        .then(res => {
+            GET('settings').then(res => res.json().then(json => {
+                assert.equal(res.status, 200);
+                return;
+            }))
+        })
+        .then(res => {
+            setTimeout(() => {
+                GET('settings')
+                .then(res => res.json().then(json => {
+                    assert.equal(res.status, 200);
+                    return;
+                }));
+            }, 3000);
+            done();
```

Comment:
Similar comments in https://github.com/plotly/plotly-database-connector/pull/180/files/e7034e717d8602fa95158af34447b5a8d4e7f712..7204ec66554f53e5732be3d1da64fcd020c0323d#r143552194 - 
`setTimout` is asynchronous and it is not a promise, so in this case `done()` is going to be called before `setTimeout` is finished. `done()` needs to be where the `return;` statement is inside `setTimeout`